tests/integration/test_mongodb.py

```diff
@@ -441,13 +442,25 @@ async def test_dao_crud_happy(use_custom_id: bool, mongodb: MongoDbFixture):
 
     assert obtained_hits == {resource_updated, resource3}
 
+    # perform a search using values with non-standard data types
+    # (note that in this case we need to serialize these values manually):
+    mapping = {
+        id_field: str(getattr(resource, id_field)),
+        "field_d": resource.field_d.isoformat(),
+    }
```

Comment:
Can we apply the same SerDe behavior from `_default_json_serializer()` to mongodb query mappings before running the query? That would enable consistent behavior for read and write. Otherwise it's not intuitive that you can store an object with a datetime/uuid but have to search with the manually serialized values.

Reply:
Yes, that's not intuitive, that's why I at least wanted to document this in a test. But I think you're right, we should change this behavior, since otherwise switching to BSON would be a breaking change, because then manual conversion will not be necessary.